commands/scanpullrequests.go

```diff
@@ -2,51 +2,58 @@ package commands
 
 import (
 	"context"
-	"errors"
-	"fmt"
 	"sort"
 	"strings"
 
 	"github.com/jfrog/frogbot/commands/utils"
 	"github.com/jfrog/froggit-go/vcsclient"
+	clientLog "github.com/jfrog/jfrog-client-go/utils/log"
 )
 
 type ScanAllPullRequestsCmd struct {
 }
 
-func (cmd ScanAllPullRequestsCmd) Run(params *utils.FrogbotParams, client vcsclient.VcsClient) error {
-	return scanAllPullRequests(params, client)
+func (cmd ScanAllPullRequestsCmd) Run(configAggregator utils.FrogbotConfigAggregator, client vcsclient.VcsClient) error {
+	for _, config := range configAggregator {
+		err := scanAllPullRequests(config, client)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
 }
 
 // Scan pull requests as follows:
 // a. Retrieve all open pull requests
 // b. Find the ones that should be scanned (new PRs or PRs with a 're-scan' comment)
 // c. Audit the dependencies of the source and the target branches.
 // d. Compare the vulnerabilities found in source and target branches, and show only the new vulnerabilities added by the pull request.
-func scanAllPullRequests(params *utils.FrogbotParams, client vcsclient.VcsClient) (err error) {
-	openPullRequests, err := client.ListOpenPullRequests(context.Background(), params.RepoOwner, params.Repo)
+func scanAllPullRequests(repo utils.FrogbotRepoConfig, client vcsclient.VcsClient) (err error) {
+	openPullRequests, err := client.ListOpenPullRequests(context.Background(), repo.RepoOwner, repo.RepoName)
 	if err != nil {
-		return
+		return err
 	}
-	var errorList []string
 	for _, pr := range openPullRequests {
-		shouldScan, e := shouldScanPullRequest(params, client, int(pr.ID))
-		if e == nil && shouldScan {
-			e = downloadAndScanPullRequest(pr, params, client)
-		}
-		// If error, save it and continue to the next PR.
+		shouldScan, e := shouldScanPullRequest(repo, client, int(pr.ID))
 		if e != nil {
-			errorList = append(errorList, fmt.Sprintf("scanning pull request from '%s' to '%s' failed:\n%s", pr.Source, pr.Target, e.Error()))
+			err = e
+			clientLog.Error(e)
+		}
+		if shouldScan {
+			e = downloadAndScanPullRequest(pr, repo, client)
+			// If error, log it and continue to the next PR.
+			if e != nil {
+				err = e
+				clientLog.Error(e)
+			}
 		}
 	}
-	if len(errorList) > 0 {
-		err = errors.New(strings.Join(errorList, "\n"))
-	}
-	return
+	return err
```

Comment:
If the the last iteration of the above loop generated a nil error, and the iteration before it generated a non nil error, this function still returns nil. Is this the behavior we'd like?
I giess we'd like an error to be returned if at least one iretation generated an error.

Reply:
What would be preferable:

1. Whenever we receive an error from a pull request, we assert.
2. Aggregate all of the errors from all of the pull requests and return them at the end of the loop.